src/v2/conversions.rs

```diff
@@ -3134,6 +3134,134 @@ impl TryFrom<NextUpdateSequenceNumbers> for super::types::queries::NextUpdateSeq
+            successor_proof: message.successor_proof.require()?.try_into()?,
+        })
+    }
+}
+
+impl TryFrom<FinalizerRound> for super::types::block_certificates::FinalizerRound {
+    type Error = tonic::Status;
+
+    fn try_from(message: FinalizerRound) -> Result<Self, Self::Error> {
+        Ok(Self {
+            round:      message.round.require()?.into(),
+            finalizers: message
+                .finalizers
+                .into_iter()
+                .map(From::from)
+                .collect::<Vec<super::types::BakerId>>(),
+        })
+    }
+}
+
+impl TryFrom<TimeoutSignature> for super::types::block_certificates::TimeoutSignature {
+    type Error = tonic::Status;
+
+    fn try_from(message: TimeoutSignature) -> Result<Self, Self::Error> {
+        Ok(Self(message.value.try_into().map_err(|_| {
+            tonic::Status::invalid_argument("Malformed TimeoutSignature.")
+        })?))
+    }
+}
+
+impl TryFrom<TimeoutCertificate> for super::types::block_certificates::TimeoutCertificate {
+    type Error = tonic::Status;
+
+    fn try_from(message: TimeoutCertificate) -> Result<Self, Self::Error> {
+        Ok(
+            Self {
+                round:                  message.round.require()?.into(),
+                min_epoch:              message.min_epoch.require()?.into(),
+                qc_rounds_first_epoch:
+                    message
+                        .qc_rounds_first_epoch
+                        .into_iter()
+                        .map(TryFrom::try_from)
+                        .collect::<Result<
+                            Vec<super::types::block_certificates::FinalizerRound>,
+                            tonic::Status,
+                        >>()?,
+                qc_rounds_second_epoch:
+                    message
+                        .qc_rounds_second_epoch
+                        .into_iter()
+                        .map(TryFrom::try_from)
+                        .collect::<Result<
+                            Vec<super::types::block_certificates::FinalizerRound>,
+                            tonic::Status,
+                        >>()?,
+                aggregate_signature:    message.aggregate_signature.require()?.try_into()?,
+            },
+        )
+    }
+}
+
+impl TryFrom<BlockCertificates> for super::types::block_certificates::BlockCertificates {
+    type Error = tonic::Status;
+
+    fn try_from(message: BlockCertificates) -> Result<Self, Self::Error> {
+        Ok(Self {
+            quorum_certificate:       message
+                .quorum_certificate
+                .map(TryInto::try_into)
+                .transpose()?,
+            timeout_certificate:      message
+                .timeout_certificate
+                .map(TryInto::try_into)
+                .transpose()?,
+            epoch_finalization_entry: message
+                .epoch_finalization_entry
+                .map(TryInto::try_into)
```

Comment:
Not that it matters, but is there a reason for using `try_into` here when you used `try_from` elsewhere?